### DISCUSSION TITLE
Backup: Clarification for incremental backup in documentation

### DISCUSSION THREAD

**Discussion by bwmeyer** [ravendb/ravendb#15308] [General]
I was seeing missing incremental backup files.
After investigating the issue, I could read in the log, that the incremental backup was not executed, because there was no change in the database since the last backup.

Unfortunately I couldn't find a hint about this in the documentation.
I think it would be helpful if you could add this to the backup chapter.

Also the hint about the retention policy and incremental backup seems not to be listet in the documentation:

> A Full Backup that is older than the specified retention time will be deleted by RavenDB server.
> If Incremental Backups exists, the Full Backup, and its incrementals, are removed only if the last incremental is older than the defined retention time.

**Comment by ayende** [comment]
If you have an hourly incremental backup, and nothing changed, we will do nothing.
There isn't anything _to_ backup.

What hint do you refer to?

**Reply by bwmeyer** [reply]
The hint is displayed in the studio when hovering the (i) icon, but it is nowhere in the documentation.
I think all these information should be "searchable" in the documentation.

**Reply by ayende** [reply]
Did you see this:

https://ravendb.net/docs/article-page/5.4/csharp/server/ongoing-tasks/backup-overview#retention-policy

**Reply by bwmeyer** [reply]
Yes, but there is no information about how **incremental** backups are handled at all.

**Reply by reebhub** [reply]
the Studio documentation explains incremental backups [here](https://ravendb.net/docs/article-page/5.4/csharp/studio/database/tasks/backup-task#scheduling-backups)

**Reply by bwmeyer** [reply]
Yes, but no information about **incremental retention policy**

So these to sentence are missing in the documentation:

> If Incremental Backups exists, the Full Backup, and its incrementals, are removed only if the last incremental is older than the defined retention time.

> An hourly incremental backup will only be proceeded, when data has changed.

**Reply by ayende** [reply]
There is no such concept as incremental backup policy.
Incremental backups are considered part of the previous full backup (since there is on meaning to restoring without it).

**Reply by bwmeyer** [reply]
Ok, one last try:

In the current online documentation, there is no description on how purging backups will be affected by existing incremental backups and the age settings.
This information is already available in the studio by hovering the (i) icon:
> If Incremental Backups exists, the Full Backup, and its incrementals, are removed only if the last incremental is older than the defined retention time.

Could you please just add this sentence also to the current online documentation?

Also it would be helpful to add a hint in the current online documentation about hourly incremental backup will not proceed, if there is no data change.

**Reply by reebhub** [reply]
Thanks
We'll add the Studio message to the documentation.
regarding the data change, did you notice the paragraph starting with "Incremental Backup will only back up the delta (changes made) of the data since the last backup that has occurred"?

**Comment by bwmeyer** [comment]
Yes, but this will not indicate that no backup file will be produced on an empty delta.
My understanding of this is, that an incremental backup will only contain the delta no matter what the delta is.
So an empty delta will still produce a backup file.

So it is not possible to check if the backuptask is working correctly by monitoring the files.
Assuming that the database has not changed in the past 12 hours, the last backupfile will be 12 hours old.
But this could also indicate, that the backup files are deleted, or the backuptask is failing since 12 hours.

It is also hard to see, if a backupset is complete.
When a single incremental file is deleted from a backup set, the backup will become incomplete.
A simple validation of file count is not possible in this case.

When you have stored the files elsewhere and you have to recover, you won't be able to analyze if the missing files are because of no data change or there was a failure on storing the files elsewhere.

As a workaround i will just update the database on a timely manner.
But at least this behavior should be documented.

**Reply by Danielle9897** [reply]
Created issue:
https://issues.hibernatingrhinos.com/issue/RDoc-2313/Incremental-Backups-Update-info
